midas2root/run_convert_to_root.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Convert and merge root output
+# Usage: ./run_convert_to_root.sh [run #]
+
+# cmd=/home/midptf/online/ptf-online-converter/midas2root_WORKING/midas2root_ptf.exe
+cmd=/home/midptf/online/converter_clean/midas2root/midas2root_ptf.exe
+
+iRun=$1
+strRun=$(printf "%05d" $iRun)
+echo "Producing root output for Run${strRun}..."
+
+cwdir=`pwd`
+
+outdir=$cwdir
+#outdir=/home/midptf/packages/rootana/libAnalyzer
+
+tmpdir=$outdir/tmp_conv_`date '+%s'`
+mkdir $tmpdir
+cd $tmpdir
+
+echo "moved to ${PWD}"
+export RETVAL='0'
+
+for ifile in $( ls ~/online/data/run${strRun}sub???.mid.gz )
+do
+
+  filename=${ifile##*/}
+  basename=${filename%.*}
+
+  echo "Filename:" $filename
+  echo "Basename:" $basename
+    
+
+
+  $cmd $ifile $RETVAL $1
+  RETVAL="$?"
+  echo "moving output000${strRun}.root to tmp${basename}.root" 
```

Comment:
This seems a reasonable solution for now.  I agree in the future we should try to do something better.

Does it work if you just try doing

midas2root_ptf.exe ~/online/data/run000sub*gz
?

Reply:
I tried this in my version with the changes and in a version without my changes. It encountered a segmentation fault in the second file in both cases.